lib/ui5Framework/maven/Installer.js

```diff
@@ -0,0 +1,497 @@
+import path from "node:path";
+import {mkdirp} from "../../utils/fs.js";
+import fs from "graceful-fs";
+import _StreamZip from "node-stream-zip";
+const StreamZip = _StreamZip.async;
+import {promisify} from "node:util";
+import Registry from "./Registry.js";
+import AbstractInstaller from "../AbstractInstaller.js";
+import _rimraf from "rimraf";
+const rimraf = promisify(_rimraf);
+const stat = promisify(fs.stat);
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+const rename = promisify(fs.rename);
+const rm = promisify(fs.rm);
+import {getLogger} from "@ui5/logger";
+const log = getLogger("ui5Framework:maven:Installer");
+const mvnTimestampRegex = /^(\d{4})(\d\d)(\d\d)(\d\d)(\d\d)(\d\d)$/;
+
+const CACHE_TIME = 32400000; // 9 hours
+
+class Installer extends AbstractInstaller {
+	/**
+	 * @param {object} parameters Parameters
+	 * @param {string} parameters.cwd
+	 * @param {string} parameters.ui5HomeDir UI5 home directory location. This will be used to store packages,
+	 * metadata and configuration used by the resolvers.
+	 * @param {string} parameters.snapshotEndpointUrl Maven repository URL.
+	 * 	Example: <code>https://registry.corp/vendor/build-snapshots/</code>
+	 * @param {string} [parameters.cacheMode="default"] Can be "default" (cache everything, invalidate after 9 hours),
```

Comment:
might be worth to create an enum for `cacheMode` instead of string and explanation